File: src/Java/servicebus/src/main/java/com/myservicebus/ConsumerRegistry.java
package com.myservicebus;

import java.util.ArrayList;
import java.util.List;

public class ConsumerRegistry {
    private final List<ConsumerDefinition<?, ?>> definitions = new ArrayList<>();

    public <TConsumer, TMessage> void register(Class<TConsumer> consumer, Class<TMessage> messageType) {
        definitions.add(new ConsumerDefinition<>(consumer, messageType));
    }

    public List<ConsumerDefinition<?, ?>> getAll() {
        return definitions;
    }
}

File: src/MyServiceBus/ConsumerRegistry.cs
namespace MyServiceBus;

public class ConsumerRegistry : IConsumerRegistry
{
    private readonly List<Type> _consumers = new();

    public void Register(Type consumerType)
    {
        _consumers.Add(consumerType);
    }

    public IReadOnlyList<Type> GetAll() => _consumers;
}